snow/engine/snowman/transitive_test.go

```diff
@@ -2921,3 +2921,108 @@ func TestEngineApplyAcceptedFrontierInQueryFailed(t *testing.T) {
 
 	require.Equal(choices.Accepted, blk.Status())
 }
+
+func TestEngineRepollsMisconfiguredSubnet(t *testing.T) {
```

Comment:
I added comments. I'm hoping to do a large refactor of both this code and the tests... When we do that hopefully all of this will be able to be cleaned up.